presto-common/src/main/java/com/facebook/presto/common/type/StandardTypes.java

```diff
@@ -37,6 +37,7 @@
     public static final String INTERVAL_DAY_TO_SECOND = "interval day to second";
     public static final String INTERVAL_YEAR_TO_MONTH = "interval year to month";
     public static final String TIMESTAMP = "timestamp";
+    public static final String TIMESTAMP_MICROSECONDS = "timestamp microseconds";
```

Comment:
`TIMESTAMP_WITH_MICROSECONDS `

Reply:
Should I rename the Type to TIMESTAMP_WITH_MICROSECONDS as well ? 

I am mixed on WITH though. The timezone is an additional info, so with is appropriate. Here the precision is different from Milliseconds and we want to represent it. But this is not exposed in grammar, when we expose it in SQL grammar, we need to be cautious on back ward compat though.

Reply:
`TIMESTAMP` probably can stay with `timestamp` for backward compatibility. Not so keen on "with" or without "with"~~  Either way might just do.

Reply:
Absolutely, TIMESTAMP should not change for MILLISECONDS precision. Removed with for now, but can change it later if required.